youtube_dl/extractor/cda.py

```
@@ -0,0 +1,80 @@
+    _TESTS = [
+        {
+            'url': 'http://www.cda.pl/video/5749950c',
+            'md5': '6f844bf51b15f31fae165365707ae970',
+            'info_dict': {
+                'id': '5749950c',
+                'ext': 'mp4',
+                'height': 720,
+                'title': 'Oto dlaczego przed zakrętem należy zwolnić.',
+                'duration': 39
+            }
+        },
+        {
+            'url': 'http://www.cda.pl/video/57413289',
+            'md5': 'a88828770a8310fc00be6c95faf7f4d5',
+            'info_dict': {
+                'id': '57413289',
+                'ext': 'mp4',
+                'title': 'Lądowanie na lotnisku na Maderze',
+                'duration': 137
+            }
+        }
+    ]
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        webpage = self._download_webpage('http://ebd.cda.pl/0x0/' + video_id, video_id)
+
+        if 'Ten film jest dostępny dla użytkowników premium' in webpage:
+            raise ExtractorError('This video is only available for premium users.', expected=True)
+
+        title = self._html_search_regex(r'<title>(.+?)</title>', webpage, 'title', fatal=False)
+
+        def _get_format(page):
+            unpacked = decode_packed_codes(page)
+            duration = self._search_regex(r"duration:\\'(.+?)\\'", unpacked, 'duration', default=None)
+            format_id = None
+            height = None
+
+            m = re.search(r'<a data-quality="(?P<format_id>[^"]+)" href="[^"]+" class="quality-btn quality-btn-active">(?P<height>[0-9]+)p<\/a>', page)
+            if m:
+                format_id = m.group('format_id')
+                height = int(m.group('height'))
+
+            return {
+                'duration': parse_duration(duration),
+                'format_id': format_id,
+                'height': height,
+                'url': self._search_regex(r"url:\\'(.+?)\\'", unpacked, 'url')
```

Yes, it will. `_get_format` is to be called on well-formed pages if this is not the case the code should be fixed elsewhere instead of hiding error here.